### PR TITLE
Mapmos test

### PR DESCRIPTION
**Pull request by tgiroux**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have read and agree to abide by the [Code of Conduct](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/Code-Of-Conduct.md)
- [ x I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [x] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [x] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [x] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

**Review by kaitlyndlee — DISMISSED** [review]
Looks good!

**Comment by tgiroux** [comment]
Don't merge. None of my changes to mapmos/main.cpp are in this PR yet, whoops

File: isis/src/base/apps/mapmos/mapmos.cpp [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+    // Get the track flag
+    bool bTrack = ui.GetBoolean("TRACK");
+    m.SetTrackFlag(bTrack);
+
+    // Gets the input file along with attributes
+    QString sInputFile = inCube->fileName();
+
+    ProcessMosaic::ImageOverlay overlay = ProcessMosaic::StringToOverlay( ui.GetString("PRIORITY") );
+
+    if (overlay == ProcessMapMosaic::UseBandPlacementCriteria) {
```
Comment (acpaquette) [inline isis/src/base/apps/mapmos/mapmos.cpp]: Could we change this from `ProcessMapMosaic` to `ProcessMosaic`. Makes it less confusing about which class has the enum.

**Review by acpaquette — COMMENTED** [review]
Overall looks good, tests cover most of the gui options. :+1: Could we add one test that checks the applog for output?

File: isis/tests/FunctionalTestsMapmos.cpp [hunk 2]
```diff
@@ -24,6 +24,9 @@ TEST_F(ThreeImageNetwork, FunctionalTestMapmosOntop) {
   QVector<QString> *cube1args = new QVector<QString>(args);
   QVector<QString> *cube2args = new QVector<QString>(args);
   QVector<QString> *cube3args = new QVector<QString>(args);
+
+  std::cout << cube1map->fileName() << std::endl;
```
Comment (kaitlyndlee) [inline isis/tests/FunctionalTestsMapmos.cpp]: Do we need this cout?

Comment (tgiroux) [reply isis/tests/FunctionalTestsMapmos.cpp]: No, thanks for catching that

**Review by kaitlyndlee — APPROVED** [review]

**Pull request merged**